### PR TITLE
fix(dot/sync): add nil header checks

### PR DESCRIPTION
**Pull request by noot**
## Changes

- fix SIGABORT introduced by recent commit to development, add nil checks for now until the reason this is happening is fixed
-
-

## Tests

```
go test ./dot/sync
```

## Issues

- n/a

## Primary Reviewer

- @timwu20

### CODE REVIEW

**Review by jimjbrettj — APPROVED** [review]
Great work!

File: dot/sync/chain_sync.go [hunk 1]
```diff
@@ -711,10 +711,20 @@ func handleReadyBlock(bd *types.BlockData, pendingBlocks DisjointBlockSet, ready
 	// if we're expecting headers, validate should ensure we have a header
 	if bd.Header == nil {
 		block := pendingBlocks.getBlock(bd.Hash)
-		bd.Header = block.header
+		if block != nil {
```
Comment (EclesioMeloJunior) [inline dot/sync/chain_sync.go:714]: I would suggest to `if block == nil` then log and return, this will remove the need of `else` branch

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/sync/chain_sync.go [hunk 2]
```diff
@@ -711,9 +711,19 @@ func handleReadyBlock(bd *types.BlockData, pendingBlocks DisjointBlockSet, ready
...
 		block := pendingBlocks.getBlock(bd.Hash)
+		if block == nil {
+			logger.Criticalf("block with unknown header is ready: hash=%s", bd.Hash)
+			return
+		}
+
 		bd.Header = block.header
 	}
 
+	if bd.Header != nil {
```
Comment (timwu20) [inline dot/sync/chain_sync.go:722]: I tried this branch on my local devnet and it wasn't finalising blocks.  After closer inspection, if `bd.Header == nil` and then the header is copied, wouldn't this if condition always be true?

**Review by timwu20 — CHANGES_REQUESTED** [review]

File: dot/sync/chain_sync.go (hunk 2 above)
Comment (noot) [reply dot/sync/chain_sync.go]: sorry i messed up the if statement here when i was updating it, should have been `== nil`, please pull again and retry!

**Pull request merged**